x-pack/plugins/security/server/routes/api_keys/query_aggregations.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RouteDefinitionParams } from '..';
+import type { ApiKeyAggregationsResponse } from '../../../common/model';
+import { wrapIntoCustomErrorResponse } from '../../errors';
+import { createLicensedRouteHandler } from '../licensed_route_handler';
+
+export function defineQueryApiKeysAggregationsRoute({
+  router,
+  getAuthenticationService,
+}: RouteDefinitionParams) {
+  router.post(
+    {
+      path: '/internal/security/api_key/_query_aggs',
+      validate: false,
+      options: {
+        access: 'internal',
+      },
+    },
+    createLicensedRouteHandler(async (context, _request, response) => {
+      try {
+        const esClient = (await context.core).elasticsearch.client;
+        const authenticationService = getAuthenticationService();
+
+        const areApiKeysEnabled = await authenticationService.apiKeys.areAPIKeysEnabled();
+
+        if (!areApiKeysEnabled) {
+          return response.notFound({
+            body: {
+              message:
+                "API keys are disabled in Elasticsearch. To use API keys enable 'xpack.security.authc.api_key.enabled' setting.",
+            },
+          });
+        }
+
+        const transportResponse =
+          await esClient.asCurrentUser.transport.request<ApiKeyAggregationsResponse>({
+            method: 'POST',
+            path: '/_security/_query/api_key',
```

Comment:
__question/suggestion:__ to improve performance a bit, can we reduce the set of fields ES returns with `filter_path` (I included the fields I see we use on the client side, but I'm not sure if I captured all of them or maybe we need less)?
```suggestion
            path: '/_security/_query/api_key',
            querystring: {
              filter_path: [
                'total',
                'aggregations.usernames.buckets.key',
                'aggregations.types.buckets.key',
                'aggregations.invalidated.doc_count',
                'aggregations.expired.doc_count',
                'aggregations.alertingKeys.doc_count',
                'aggregations.managedMetadata.doc_count',
              ],
            },
```